### PR TITLE
Add TUI-local file-backed execution profiles

### PR DESCRIPTION
**Pull request by harryalbert**
## Description

Before this change, agent execution profiles were backed by a set of warp drive cloud objects instead of by the settings file. This is not great, and it's especially bad for the TUI where all settings edits must go through the settings file. 

This PR updates it so that profiles are now backed by the settings file. There is a one-time migration when you first open the GUI/TUI and you have a cloud execution profile(s) but none in your settings file (where those profile(s) are moved to the settings file), and from then on out the profiles are just file based.

note: this PR is very large, but there are only a few files that actually need to be read (the rest are either tests or call signature churn):
* the tech spec
* `app/src/ai/execution_profiles/config.rs`
* `app/src/ai/execution_profiles/profiles.rs`
* `app/src/settings/ai.rs`

## Testing

- [x] I have manually tested my changes locally with `./script/run`

## Screenshots/Video

https://www.loom.com/share/b2c483c8ff4a4e60824909dd4b295046

## Agent Mode

- [x] Warp Agent Mode - This PR was created via Warp's AI Agent Mode

_This PR was generated with [Oz](https://warp.dev/oz)._

### CODE REVIEW

**Comment by harryalbert** [comment]
* **#13886** <a href="https://app.graphite.com/github/pr/warpdotdev/warp/13886?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/warpdotdev/warp/13886?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `master`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by harryalbert** [comment]
@cla-bot check

File: crates/warp_features/src/lib.rs [hunk 1]
```diff
@@ -984,13 +989,15 @@ pub const DOGFOOD_FLAGS: &[FeatureFlag] = &[
...
     FeatureFlag::PromptCacheExpiryWarning,
+    FeatureFlag::FileBackedExecutionProfiles,
     FeatureFlag::BackgroundComputerUse,
     FeatureFlag::ContextWindowUsageBreakdown,
     FeatureFlag::JupyterNotebookRendering,
     FeatureFlag::WaitForEventsParentRegistration,
     FeatureFlag::McpJsonTreeView,
     FeatureFlag::GeminiEnterprise,
     FeatureFlag::BoxDrawingGlyphs,
+    FeatureFlag::CloudAgentRunners,
```
Comment (danielpeng2) [inline crates/warp_features/src/lib.rs]: unintended side-effect?

File: app/src/ai/execution_profiles/config.rs [hunk 2]
```diff
@@ -0,0 +1,584 @@
...
+
+/// User-editable representation of one execution profile.
+///
+/// This deliberately omits `is_default_profile`; the containing map key owns
+/// that invariant. String-backed regex and UUID fields are validated while
+/// converting back to [`AIExecutionProfile`].
+#[derive(Debug, Clone, Serialize, Deserialize, schemars::JsonSchema)]
+#[serde(default)]
+struct ExecutionProfileFile {
+    /// User-facing profile name.
```
Comment (danielpeng2) [inline app/src/ai/execution_profiles/config.rs]: I have a slight preference for writing these comments like: `#[schemars(description = "User-facing profile name.")]`.

These comments are written into the JSON schema that is used by the agent and our docs. Having an explicit `schemars` annotation makes it clearer that these comments are public-facing.

**Review by danielpeng2 — APPROVED** [review]
Overall looks good, there's a couple things we could do to be extra careful:
- Make sure we test new user flows. These had some tricky edge cases when we were initially building out settings files, though it may not be as relevant as I don't think loginless users can modify AI profiles? e.g.
  - Download Warp -> start as loginless user and modify settings -> sign up for new account
  - Download Warp -> start as loginless user and modify settings -> sign in to existing account
- Write integration tests to a test bit more end-to-end, there should be examples of existing settings file integration tests.

File: crates/warp_features/src/lib.rs (hunk 1 above)
Comment (harryalbert) [reply crates/warp_features/src/lib.rs]: yeah good call

File: app/src/ai/execution_profiles/config.rs (hunk 2 above)
Comment (harryalbert) [reply app/src/ai/execution_profiles/config.rs]: oh cool, good call

**Comment by harryalbert** [comment]
> Overall looks good, there's a couple things we could do to be extra careful:
> 
> * Make sure we test new user flows. These had some tricky edge cases when we were initially building out settings files, though it may not be as relevant as I don't think loginless users can modify AI profiles? e.g.
>   
>   * Download Warp -> start as loginless user and modify settings -> sign up for new account
>   * Download Warp -> start as loginless user and modify settings -> sign in to existing account
> * Write integration tests to a test bit more end-to-end, there should be examples of existing settings file integration tests.

Yeah now that there's no free AI (or like especially, no AI w/ no login), it's not possible to hit this edge case. But will definitely add an integration test

File: app/src/ai/execution_profiles/profiles.rs [hunk 3]
```diff
@@ -201,12 +306,49 @@ impl AIExecutionProfilesModel {
...
+                }
+            });
+
+            if should_migrate_legacy_cloud_profiles {
+                if ctx.has_singleton_model::<CloudPreferencesSyncer>() {
+                    ctx.subscribe_to_model(
+                        &CloudPreferencesSyncer::handle(ctx),
+                        |me, _, event, ctx| {
+                            if matches!(event, CloudPreferencesSyncerEvent::InitialLoadCompleted) {
+                                me.migrate_settings_profiles(ctx);
```
[the remark this answers is not in the document]
Comment (harryalbert) [reply app/src/ai/execution_profiles/profiles.rs:326]: idt this is a problem
